### DISCUSSION TITLE
How to detect an incoming call

### DISCUSSION THREAD

**Discussion by 313hemant313** [ionic-team/capacitor#3992] [Q&A] [answered]
The Capacitor AppState provides status of app when the app enters and leaves the foreground, But is it possible to detect incoming call ? 

For me the main task in hand is to mute/volume down the audio that is currently playing when a phone call comes.

**Comment by jcesarmobile** [comment, marked as answer]
With App plugin you can't reliably know if the app was paused by a phone call, just to know that the app was paused, for some reason. If you only want to pause the sound when there is a phone call you should create your own plugin for detecting incoming calls.

**Reply by 313hemant313** [reply]
Created an android plugin to detect call state ('RINGING' | 'OUTGOING' | 'IDLE' | 'ON_CALL' | 'ON_HOLD').
Feel free to contribute or use.

capacitor-plugin-incoming-call
Repository: https://github.com/313hemant313/capacitor-plugin-incoming-call
Package: https://www.npmjs.com/package/capacitor-plugin-incoming-call

**iOS plugin added.
